staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -1180,6 +1180,13 @@ type ManagedFieldsEntry struct {
 	// FieldsV1 holds the first JSON version format as described in the "FieldsV1" type.
 	// +optional
 	FieldsV1 *FieldsV1 `json:"fieldsV1,omitempty" protobuf:"bytes,7,opt,name=fieldsV1"`
+
+	// Subresource is the name of the subresource used to update that object, or
+	// empty string if the object was updated through the main resource. The
+	// value of this field is used to distinguish between managers, even if they
+	// share the same name. For example, a status update will be distinct from a
+	// regular update using the same manager name.
+	Subresource string `json:"subresource,omitempty" protobuf:"bytes,8,opt,name=subresource"`
```

Comment:
How would this work with a subresource that accepts two different versions, as is currently being added for pod/evict? (fortunately, that isn't the kind of subresource that matters for this)

Reply:
If subresource is non-empty, does it change the meaning of APIVersion? Does APIVersion remain about the parent resource's kind, or does it now refer to the subresource's kind?

Reply:
Yeah, it's always about the parent resource. We do convert scale objects to parent resource version objects (and back) but the managedFields are always in parent resource version (otherwise we wouldn't be able to convert later)

Reply:
OK, that's right, because you've converted from whatever format the subresource was in already. OK. Comment could maybe mention this.